Fix aws_ssm when Powershell uses CLIXML

matpompili opened the pull request:
##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->

This PR adds the code suggested by @PatrickV-code on #1952
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->

> In short, connection to Windows based EC2 instances using aws_ssm connection and using the become option of runas fails every job.
> The job itself actually finishes, but as Powershell 5 adds additional wrapping with CLIXML information.
> Which makes the output unreadable for Ansible.

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Bugfix Pull Request

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->
aws_ssm

tremble commented:
recheck

matpompili commented:
Hi @tremble, anything else I can do for this PR?

markuman commented:
regate

alinabuzachis commented:
regate

alinabuzachis commented:
regate

markuman commented:
regate

markuman commented:
> 2024-06-28 19:10:40.087341 | controller |     "msg": "timed out waiting for ping module test: 'ping'"

markuman commented:
regate

alinabuzachis commented:
regate

markuman commented:
regate

matpompili commented:
Hi @markuman @alinabuzachis, is there something I need to change for the build to succeed?

markuman commented:
> Hi @markuman @alinabuzachis, is there something I need to change for the build to succeed?

I'm not sure. Error message is 

```
TASK [wait_for_connection] *****************************************************
task path: /home/zuul-worker/.ansible/collections/ansible_collections/community/aws/tests/integration/targets/connection/test_connection.yml:14
fatal: [linux_i-0b10c2ed71e4a0682]: FAILED! => {
     "changed": false,
     "elapsed": 101,
     "msg": "timed out waiting for ping module test: 'ping'"
}
```

gateing CI failed, while the check for the PR  succeeded.

markuman commented:
recheck

matpompili commented:
@markuman @alinabuzachis I merged the latest commits from main into this PR, hopefully that fixes the checks?

matpompili commented:
@markuman @alinabuzachis I don't follow why the build doesn't succeed. 

The edits work when I use them on my AWS instance.

It seems like something was changed between https://github.com/ansible-collections/community.aws/pull/2080#issuecomment-2079959483 and https://github.com/ansible-collections/community.aws/pull/2080#issuecomment-2196930063, but not the code of this PR.

tenpei-on commented:
This PR looks promising for resolving the CLIXML handling issue in aws_ssm with PowerShell. Our team has been eagerly waiting for this fix as it would significantly streamline our workflows. Thank you for the continuous efforts! Looking forward to seeing this merged soon. If there’s any way we can help or test further, please let us know.

markuman commented:
recheck

alinabuzachis commented:
recheck

alinabuzachis commented:
recheck

matpompili commented:
@markuman @alinabuzachis I updated the PR to include the latest changes in `main`, but the CI still failed on the `ping` test. I tried commenting out the code I added, to see if in some obscure way that was the issue, but the CI failed on the same exact manner.

I am at a loss here, we really need this change merged, and I don't know what I can do to get that workflow to pass. Suggestions?

abikouo commented:
recheck
